packages/web-frontend/src/containers/MapOverlayBar/Control.js

```diff
@@ -96,11 +145,25 @@ export const Control = ({
     }
   }, [isExpanded, setIsExpanded]);
 
+  const reorderedSelectedMapOverlays = useMemo(() => {
+    const results = [];
+    if (!pinnedOverlay) {
+      return selectedMapOverlays;
+    }
+
+    selectedMapOverlays.forEach(overlay => {
+      if (overlay.mapOverlayCode === pinnedOverlay) {
+        results.unshift(overlay);
+      } else results.push(overlay);
+    });
+
+    return results;
+  }, [selectedMapOverlays, pinnedOverlay]);
+
```

Comment:
```suggestion
  const reorderedSelectedMapOverlays = useMemo(() => {
    if (!pinnedOverlay) {
      return selectedMapOverlays;
    }

    return selectedMapOverlays.sort(first => (first.mapOverlayCode === pinnedOverlay ? -1 : 0));
  }, [selectedMapOverlays, pinnedOverlay]);
```

Optional suggestion that you could use sort to do this.

Reply:
```suggestion
  const reorderedSelectedMapOverlays = useMemo(
    () =>
      pinnedOverlay
        ? selectedMapOverlays.sort(first => (first.mapOverlayCode === pinnedOverlay ? -1 : 0))
        : selectedMapOverlays,
    [selectedMapOverlays, pinnedOverlay],
  );
```
Or even this

Reply:
Very smart way to do it! Definitely will use the last one in the code.